--- a/src/minicoredumper/corestripper.c
+++ b/src/minicoredumper/corestripper.c
@@ -1241,7 +1241,7 @@
 	snprintf(hdr.mode, sizeof(hdr.mode), "%07o", 0644);
 	snprintf(hdr.uid, sizeof(hdr.uid), "%07o", 0);
 	snprintf(hdr.gid, sizeof(hdr.gid), "%07o", 0);
-	snprintf(hdr.mtime, sizeof(hdr.mtime), "%011lo",
+	snprintf(hdr.mtime, sizeof(hdr.mtime), "%011llo",
 		 (long long)time(NULL));
 	memset(hdr.checksum, ' ', sizeof(hdr.checksum));
 	hdr.type = 'S';
